File: 0001-Two-Sum/0001.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for index, value in enumerate(nums):
            # if (target - value) in the dictionary 
            if target - value in hash:
                return [hash[target - value], index]
            hash[value] = index

File: 0001-Two-Sum/0001.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            // if value of (target - nums[i]) in the HashMap Key
            if (hash.containsKey(target - nums[i])) {
                int[] answer = { hash.get(target - nums[i]), i };
                return answer;
            }
            hash.put(nums[i], i);
        }

        return null;
    }
}